src/ir/function.rs

```diff
@@ -495,7 +495,15 @@ impl FunctionSig {
         } else {
             ty.ret_type().ok_or(ParseError::Continue)?
         };
-        let ret = Item::from_ty_or_ref(ty_ret_type, cursor, None, ctx);
+
+        let ret = if ctx.is_target_wasm32() && is_constructor {
```

Comment:
Can we switch the order here, `is_constructor` is cheaper to check, and should be false most of the time.

Also, can we add a test for this? Should be doable passing `--target` in the `bindgen-args` like `tests/headers/win32-thiscall_1_0.hpp` and others do.

Reply:
Added a commit which addresses both of these comments.